test_util/cluster_api.py

```diff
@@ -201,55 +202,50 @@ def __init__(self, dcos_uri, masters, public_masters, slaves, public_slaves,
 
         # URI must include scheme
         assert dcos_uri.startswith('http')
-        self.scheme = urlparse(dcos_uri).scheme
+        parse_result = urlparse(dcos_uri)
+        self.scheme = parse_result.scheme
+        self.dns_host = parse_result.netloc.split(':')[0]
 
         # Make URI never end with /
         self.dcos_uri = dcos_uri.rstrip('/')
 
-    @staticmethod
-    def _marathon_req_headers():
-        return {'Accept': 'application/json, text/plain, */*'}
+        # Generate request functions
+        for method in self.request_methods:
```

Comment:
https://docs.python.org/3/library/functools.html#functools.partialmethod might be useful.

Could then do something like:
```
get = partialmethod(cluster_request_method_name, 'get')
post = partialmethod(cluster_request_method_name, 'post')
```
on the class for the ones we actually use / care about.

currently this adds a bunch of member functions which are very hard to trace back to their origins (introspection tools in editors and any sort of static analysis will likely fail hard)